RuleDocumentation/PossibleIncorrectComparisonWithNull.md

```diff
@@ -41,6 +41,7 @@ function Test-CompareWithNull
 ## Try it Yourself
 
 ``` PowerShell
-if (@() -eq $null) { 'true' } else { 'false' }  # Returns false
-if ($null -ne @()) { 'true' } else { 'false' }  # Returns true
+# Both expressions below return 'false' because the comparison does not return an object and therefore the if statement always falls through:
+if (@() -eq $null) { 'true' }else { 'false' }
```

Comment:
```suggestion
if (@() -eq $null) { 'true' } else { 'false' }
```

Reply:
FWIW, this is due to how `-eq` works with collections. Here's an illuminating example
```powershell
PS> 1,2,3,1,2 -eq $null
PS> 1,2,3,1,2 -eq 1    
1
1
PS> (1,2,3,1,2 -eq $null).count            
0
PS> (1,2,$null,3,$null,1,2 -eq $null).count
2
```
for each element in the collection, the comparison with the RHS is done, and where true, that element in the collection is returned.

Reply:
Yeah I think including an example like this might be more helpful